### PR TITLE
[14.0][MIG] sale_order_product_recommendation: Migration to 14.0

### PR DESCRIPTION
**Pull request by ShivamS-SerpentCS**

### CODE REVIEW

File: sale_order_product_recommendation/__manifest__.py [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
-    "version": "13.0.2.1.0",
+    "version": "14.0.1.0.0",
     "category": "Sales",
     "website": "https://github.com/OCA/sale-workflow",
     "author": "Tecnativa, Odoo Community Association (OCA)",
     "license": "AGPL-3",
     "application": False,
     "installable": True,
-    "depends": ["sale"],
+    "depends": ["sale_management"],
```
Comment (rousseldenis) [inline sale_order_product_recommendation/__manifest__.py:14]: Why this change ?

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def _generate_recommendations(self):
             key=lambda res: (res["product_id_count"], res["qty_delivered"]),
             reverse=True,
         )
-        found_dict = {l["product_id"][0]: l for l in found_lines}
+        found_dict = {L["product_id"][0]: L for L in found_lines}
```
Comment (rousseldenis) [inline sale_order_product_recommendation/wizards/sale_order_recommendation.py:110]: If warning was to change, not to go for caps. :smile: 'line' is better

**Comment by francesco-ooops** [comment]
@ShivamS-SerpentCS any plans to finish/merge this one?
Thanks!

**Comment by pedrobaeza** [comment]
Please include #1980

**Comment by willytal** [comment]
Please include https://github.com/OCA/sale-workflow/pull/2065

**Comment by rousseldenis** [comment]
@ShivamS-SerpentCS Do you plan to continue this ?
